### ISSUE TITLE
Vanish/Shadowmeld behavior

### ISSUE THREAD

**Issue by RandomReportForYou** [azerothcore/azerothcore-wotlk#2668]
##### SMALL DESCRIPTION:
Vanish is granting a small window of immunity after being used, while it should break active movement impairments, it shouldn't make you immune to them.

##### EXPECTED BLIZZLIKE BEHAVIOUR:
After using Vanish, Piercing Howl should affect the rogue and break stealth.

##### CURRENT BEHAVIOUR:
After using Vanish, there's a brief window of immunity (0.5s +-), during which using Piercing Howl against the rogue will show a IMMUNE message, while the rogue has no immunity against it, other spells like Demoralizing Shout or Thunderclap will still work, I haven't tested with other classes spell neither.
Vanish https://wowgaming.altervista.org/aowow/?spell=26889
Piercing Howl https://wowgaming.altervista.org/aowow/?spell=12323
Demoralizing Shout https://wowgaming.altervista.org/aowow/?spell=47437

##### STEPS TO REPRODUCE THE PROBLEM:
1) Use Vanish in a rogue
2) Quickly (must be less than a second) use Piercing Howl
3) Rogue is immune to Piercing Howl

##### EXTRA NOTES:
Tested using level 80 characters several times, 100% reproduction rate and if you take about a second or more the rogue won't be immune, while it might not seem much, in a pvp match, specially arena, this means the rogue getting away and it's punishing quick reflexes.
It might be worth testing other slows, snares or roots like Frost Nova, or Earthbind Totem with the Storm, Earth and Fire talent.

##### BRANCH(ES):
master

##### AC HASH/COMMIT:
e1ee77e173e177d16b676ac8fac137eab3bfe818

##### OPERATING SYSTEM:
TBD

##### MODULES:
none

##### OTHER CUSTOMIZATIONS:
none

**Comment by Degenesis** [comment]
https://wowwiki.fandom.com/wiki/Vanish
"Later on you could even negate instant cast spell effects due to short amount of time of immunity Vanish provided, which was added to the game because of rogue players complains about Vanish being randomly broken. This was changed in early Cataclysm after Vanish was reworked into providing unbreakable 3 second Stealth. "

https://www.engadget.com/2009/10/07/encrypted-text-vanish-on-the-patch-3-3-ptr/
"Since this article appeared, the patch notes were edited to reflect a 0.5 second immunity, which coincides with the results seen here."

Not a bug, just a feature of WotLK.

**Comment by RandomReportForYou** [comment]
> https://wowwiki.fandom.com/wiki/Vanish
> "Later on you could even negate instant cast spell effects due to short amount of time of immunity Vanish provided, which was added to the game because of rogue players complains about Vanish being randomly broken. This was changed in early Cataclysm after Vanish was reworked into providing unbreakable 3 second Stealth. "
> 
> https://www.engadget.com/**2009/10/07**/encrypted-text-vanish-on-the-patch-3-3-ptr/
> "patch notes reflect a 0.5 second immunity, which coincides with the results seen here."
> 
> Not a bug, just a feature of WotLK.

That only proves it being on the PTR though, not on the live patch, just like this wowhead comment says:

>  https://www.wowhead.com/spell=1856/vanish#comments:0
> Supahtoons on 2009/10/24 (Patch 3.2.2)
>  As of the new build for the PTR on **10/23/09**, the Vanish change has been reverted.

which was in response to: 

> lovechia on **2009/09/29** (Patch 3.2.2) 
> > We have a change for Vanish in place for 3.3. You will get to try it out soon (tm). As promised, if it proves a significant buff to rogues, we may have to compensate elsewhere. Just because it hasn't worked as intended doesn't mean it will be balanced when it does.

> Source: http://forums.worldofwarcraft.com/thread.html?topicId=20136596247&pageNo=3&sid=1#44

This leads me to believe that the change was made and went to the PTR, but never made it to the live patch, moreso there's no mention of any change to Vanish in the patch notes here.

https://wowwiki.fandom.com/wiki/Patch_3.3.0a
https://wowwiki.fandom.com/wiki/Patch_3.3.2_(undocumented_changes)
https://wowwiki.fandom.com/wiki/Patch_3.3.2
https://wowwiki.fandom.com/wiki/Patch_3.3.3_(undocumented_changes)
https://wowwiki.fandom.com/wiki/Patch_3.3.3
https://wowwiki.fandom.com/wiki/Patch_3.3.3a_(undocumented_changes)
https://wowwiki.fandom.com/wiki/Patch_3.3.3a
https://wowwiki.fandom.com/wiki/Patch_3.3.5_(undocumented_changes)
https://wowwiki.fandom.com/wiki/Patch_3.3.5
https://wowwiki.fandom.com/wiki/Patch_3.3.5a_(undocumented_changes)
https://wowwiki.fandom.com/wiki/Patch_3.3.5a

**Comment by BarbzYHOOL** [comment]
And for the curious out there, what is the best concept: 0.5s immunity or not?

**Comment by Degenesis** [comment]
It was changed in Cata to grant unbreakable stealth for 3s, and for good reason.

**Comment by BarbzYHOOL** [comment]
3seconds?? rofl

**Comment by gengarshadowball** [comment]
I would leave it as it is since everyone is probably used to it and that's probably how it works on most of cores making this "Blizzlike".

**Comment by RandomReportForYou** [comment]
> It was changed in Cata to grant unbreakable stealth for 3s, and for good reason.

That's not a good argument though, because a lot of other things were changed for cata, and not all of those changed are here in WotLK for the different classes.

> I would leave it as it is since everyone is probably used to it and that's probably how it works on most of cores making this "Blizzlike".

So if a bug is present on most cores, that makes the bug "blizzlike" and should remain there without being fixed huh.

Your comments makes it seem like you're acknowledging that this is indeed a bug, but a bug you want to maintain because: "they buffed this skill in cata" or "everyone is probably used to it". Would this mean we can add the "Confirmed" label then? :p

**Comment by BarbzYHOOL** [comment]
no, these reasons are not valid, let's wait for other people's (blizzlike experts?) input because if we change something and then later on have to revert it, it's manpower wasted :)

**Comment by gengarshadowball** [comment]
> > It was changed in Cata to grant unbreakable stealth for 3s, and for good reason.
> 
> That's not a good argument though, because a lot of other things were changed for cata, and not all of those changed are here in WotLK for all the different classes.
> 
> > I would leave it as it is since everyone is probably used to it and that's probably how it works on most of cores making this "Blizzlike".
> 
> So if a bug is present on most cores, that makes the bug "blizzlike" and should remain there without being fixed huh.
> 
> Your comments makes it seem like you're acknowledging that this is indeed a bug, but a bug you want to maintain because: "they buffed this skill in cata" or "everyone is probably used to it". Would this mean we can add the "Confirmed" label then? :p

Let me clarify, lets leave THIS as it is since:
- you don't have proof of this behaviour with videos or sniffs or footage
- anyone can come here clamining this is the correct behaviour when we could be introducing a bug
-unclear comments on websites are miss leading

THIS specific behaviour should remain as it is since you can't go back on time to check how Blizzard did it. You can also wait until blizzard invents a WOTLK realm on official servers.

**Comment by Si1ker** [comment]
I think @gengarshadowball and @Degenesis got the point here and until no further (blizzlike) evidence is provided, it should be labeled as Not Confirmed (as the "bug" isn't confirmed). It is how it is for a reason.

**Comment by Endeffect-AT** [comment]
i agree with @gengershadowball as long as there is no proof that this is a bug it should not be changed.

**Comment by RandomReportForYou** [comment]
I'm trying to find any video which either shows the immunity or the lack of it, but it has been a nightmare to do, and I'm reporting this as a bug because besides the PTR note, there's no evidence at all about this change in Vanish sticking to a live patch, even though the other PTR changes did show in the proper patch notes like the dual wield at level 1. 
Please do help in finding any evidence to either proposition, I found a channel with a bunch of holy/war arena videos, which i have been using to try and find them against a rogue and them using Piercing Howl after Vanish, but no luck so far.

> https://www.youtube.com/user/Belligerent23/videos?view=0&sort=da&flow=grid

Also I found this one, but it's not as clear as I would like, still I'm gonna leave it here:

> https://youtu.be/OP63cb5cHf0?t=532

BTW, If you use Vanish and cancel the stealth, the immunity window remains even though you don't have vanish, so what matters is that you use vanish to get the immunity even if you instantly exit stealth.

**Comment by BarbzYHOOL** [comment]
> I'm trying to find any video which either shows the immunity or the lack of it, but it has been a nightmare to do, and I'm reporting this as a bug because besides the PTR note, there's no evidence at all about this change in Vanish sticking to a live patch, even though the other PTR changes did show in the proper patch notes like the dual wield at level 1.
> Please do help in finding any evidence to either proposition, I found a channel with a bunch of holy/war arena videos, which i have been using to try and find them against a rogue and them using Piercing Howl after Vanish, but no luck so far.
> 
> > https://www.youtube.com/user/Belligerent23/videos?view=0&sort=da&flow=grid
> 
> Also I found this one, but it's not as clear as I would like, still I'm gonna leave it here:
> 
> > https://youtu.be/OP63cb5cHf0?t=532
> 
> BTW, If you use Vanish and cancel the stealth, the immunity window remains even though you don't have vanish, so what matters is that you use vanish to get the immunity even if you instantly exit stealth.

at least we're sure this is a bug (but who would do that ingame xD)

**Comment by Degenesis** [comment]
"fixing" this would likely upset a lot of rogue players as getting knocked right out of Vanish was a common problem that was never intended. 

I think in this case the bug is more popular than the blizzlike behavior, like the tree stumps in WSG breaking line-of-sight when they shouldn't.

**Comment by RandomReportForYou** [comment]
As I said in the original report, even though there's a brief immunity window, it doesn't make you immune to everything, Demoralizing Shout or Thunderclap do work, so in the sense of it preventing you from "_getting knocked right out of Vanish_", well, it's not doing that, it's only making you immune to **some** spells, like Piercing Howl (which is the most efficient spell to use in those conditions, as it should break stealth and slow the rogue), and it's doing it even if you leave stealth.

**Comment by Maurowin** [comment]
I've come to throw some fuel on the fire.
For my part, from my distant memory, VANISH was giving us:
- immune to any hindrance affecting movement;

I can no longer say how long that immunity lasted. 

Under no circumstances:
- Remove any dots OTHER than those affecting displacement;

All other magic dots were removed by the spell shadow cap once the player used it.

Any physical damage over time was not removed.

**Comment by Shard-MW** [comment]
> After using Vanish, Piercing Howl should affect the rogue and break stealth.

Well, nope. Casting Vanish at the same time you get hit by a direct spell always l led to avoid this recieved spell's damages and effects. 
You could avoid POM Pyro, Death Coil, Kidney Shot, Deep Freeze, Instant Lavaburst etc etc
But ofc it'll break if you ALREADY have DoT on you, and you Vanish at the same time that DoT hit you.

Please take a look on the videos that you can find on this issue : 
https://github.com/azerothcore/azerothcore-wotlk/issues/1968

Plus, 
> https://youtu.be/OP63cb5cHf0?t=532
Here stealth broke because the rogue casted something (maybe Ambush?). Look at rogue's energy before Vanish and after...

**Comment by RandomReportForYou** [comment]
@Shard-MW but you just ignored what I posted above, which in case you didn't saw was that **you can get forced out of "Vanish" with other spells** like Demoralizing Shout or Thunderclap, Vanish gives you no immunity against them.

Besides this report is about breaking stealth and slowing the rogue **after** it uses Vanish, and the immunity window that it gives that prevents it, which from what I found is unjustified as **it was only implemented on a PTR patch as far as I can find**. Also, those spells you mention are proyectiles which were casted **before** the rogue used Vanish to prevent them from **reaching** the rogue, for example it would be similar to a warrior using spell reflect on the Pyro, the Pyro is on the air towards the warrior and the warrior swaps to its 2H, losing the spell reflect buff obviously, but the Pyro still gets reflected even though the buff is no longer there, because of how the server handles those proyectiles, but in the case of vanish the proyectile goes to rogue A, and it would seem to the server that rogue A *vanished* so it no longer exist, ergo it never reaches the rogue, but if you could still see the stealthed rogue you should be able to launch another instant proyectile against it and **it should** hit, or in other words you're mixing the 0.5s immunity with the eating proyectiles mechanic, which would make sense given that the 2 spells I mentioned do break the rogue from stealth after it uses Vanish.

Finally about the video I provided, it wasn't about the rogue being forced out of stealth by a spell, it was about the immunity against piercing howl that it provided, which the rogue got afflicted by in the video, and now you could say that he wasn't stealthed and so he didn't have any immunity obviously, but as I mentioned after the video, you still have the immunity even though you're not stealthed.

**tl;dr**

> Casting Vanish at the same time you get hit by a direct spell always l led to avoid this recieved spell's damages and effects.

As far as I can tell has nothing to do with this report, because it's not being casted before or at the same time, but after.

> Here stealth broke because the rogue casted something

The video was about the rogue **NOT** being immune to piercing howl after using Vanish.

This is not about the rogue removing stuff, its about the 0.5s immunity it grants to **some** spells, like Piercing Howl.

**Comment by Shard-MW** [comment]
Alright, sorry, I misunderstood that you were speaking about spells casted **after** Vanish.

But actually, according to DBCs Vanish triggers this spell : 
https://wotlk.evowow.com/?spell=18461
https://classic.wowhead.com/spell=18461
which grants a 1 sec immunity to root and snares. 

I think that having it on Classic Wowhead is enough to prove that it has not been
> only implemented on a PTR patch

So we can say that the spell is working as intended. (Same thing for Blink btw)

> Also, those spells you mention

Cheap shot, Kidney, Deep freeze are not projectiles, but anyway it makes no sense to speak about it now haha

> The video was about the rogue NOT being immune to piercing howl after using Vanish.

Rogue canceled his Vanish (and so all auras that Vanish triggered) by casting an Ambush. That's why he got hit by Piercing Howl, because immunity was already gone.

But if you now say that, even after instant canceling stealth, rogue is still immune to snares & roots effects during 1 sec, this will be another issue.

**Comment by RandomReportForYou** [comment]
> But actually, according to DBCs Vanish triggers this spell :
https://wotlk.evowow.com/?spell=18461
https://classic.wowhead.com/spell=18461
which grants a 1 sec immunity to root and snares.

>Rogue canceled his Vanish (and so all auras that Vanish triggered) by casting an Ambush. That's why he got hit by Piercing Howl, because immunity was already gone.

It says immunity to snares and roots, yes, but you mentioned immunity to damage (lava, pyro, etc) and Piercing Howl is a Slow/Daze, not a root nor snare, so this wouldn't cover neither, and as I said, the immunity persists regardless of the rogue being or not in stealth, and yes, again, the immunity persist regardless of exiting stealth, which is why I gave the video as evidence of the lack of immunity.

> I think that having it on Classic Wowhead is enough to prove that it has not been

Also, that spell seems to be from the original vanish, while the PTR said it added something, so this should be some other mechanic from the spell (maybe removing existing cc) and the fact that it last 1 seconds, while the immunity talked here lasts 0.5s, was added at patch 3.3 and is not about root or snares.

**Comment by Shard-MW** [comment]
> but you mentioned immunity to damage (lava, pyro, etc)

I just mentionned that you AVOID damages & spell effects when you cast Vanish **at the same time/before**

> immunity persists regardless of the rogue being or **not in stealth**

I confirm that rogue is still immune to root/snares effects during 1sec **even after removing stealth**. 
I think (by my game logic) that Immunity should be removed when Stealth/Vanish is removed, but I can't confirm my though. Because, according to the video @RandomReportForYou sent, the rogue vanishes at ~8:52 and get snared at ~8:54. There is more than 1 sec difference...
I think we need more research here

**Comment by Andrei-UPB** [comment]
Snares are actually slows and other movement impairing effects that do not completely hinder the character from moving, thus any slow effect, be it a daze or a normal slow is classified as a snare.

https://wow.gamepedia.com/Snare provides more information about what CC term is used for what.

If gamepedia is not a trusted source, just enable "Target Effects" from the interface option (Esc -> Interface -> Combat Text -> Top Right Corner, "Target Effects") and try to use spells like Piercing Howl, Frostfire Bolt or whichever other slow comes in handy on a target, the message Snared will show up briefly as yellow text when the debuff is applied.

**Comment by syssneck** [comment]
Does anyone know if this is still an issue?

**Comment by Shineslippers** [comment]
Still an issue.
1. Thunder Clap, Demoralizing Shout remove stealth from rogue
2. Using Piercing Howl saying `immune'
3. Other damage abilities like frost bolt doesn't remove stealth and doesn't leave somethng in combat log

![Capture](https://user-images.githubusercontent.com/85256266/146399906-f7171798-043d-43cf-ba55-b2f03c91675a.PNG)

**Comment by laasker** [comment]
looks like the Vanish spell trigger that removes slow/root is granting immunity to slows for 1s (piercing howl is a slow)

@RandomReportForYou i dont know if its a bug or not but i was able to fix this with those steps:
1. go to spell_dbc in world db
2. change Spellid 18461 (Vanish Purge) DurationIndex from 36 to 37 (1000 milliseconds to 1 millisecond)

also because of this report i noticed Piercing Howl is not revealing players in stealth/vanish because of this PR:
https://github.com/azerothcore/azerothcore-wotlk/pull/3056/files
is it intended? i think only in cata it was supposed to not reveal stealth players

**Comment by ciippo** [comment]
Vanish/Shadowmeld had a inmunity in wotlk retail. If a projectile is flighting, you are totally inmune if u hit (pyro, death-coil,...).
Also you had inmunity against instant skills if u vanished correctly
 1. Instant cc is casted and then u vanished -> cc inmunized and vanish breaks
 2. Vanish is casted and then instant cc hit you -> cc hit rogue and vanish breaks (not inmunized)
How could cc hit you after vanish? easy, vanish wasn't instant, he had a small window of visibility where you can see the rogue vanishes (you can see stealth/vanish buff on target bar). This allowed sap vanishes too (This is bugged here i think).

Quality is soo bad (very old video, but you can see buff vanish/stealth on target rogue, and how he can vanish blinds and stealth is broken):

https://www.youtube.com/watch?v=spH5R2pTBFU

0:17 Buffs on target rogue are displayed
0:22 He cast blind AFTER enemy rogue vanishes(see buffs again) and cc is applied
0:48 He vanishes when blind is being casted (vanish breaks and he is inmunized)
1:31 Blind vanish
1:46 Vanish blind
2:07 inmune rocket gloves
2:36 Sap vanish
3:13 Vanish blind

**Comment by laasker** [comment]
> Vanish/Shadowmeld had a inmunity in wotlk retail. If a projectile is flighting, you are totally inmune if u hit (pyro, death-coil,...). Also you had inmunity against instant skills if u vanished correctly
> 
> 1. Instant cc is casted and then u vanished -> cc inmunized and vanish breaks
> 2. Vanish is casted and then instant cc hit you -> cc hit rogue and vanish breaks (not inmunized)
>    How could cc hit you after vanish? easy, vanish wasn't instant, he had a small window of visibility where you can see the rogue vanishes (you can see stealth/vanish buff on target bar). This allowed sap vanishes too (This is bugged here i think).
> 
> Quality is soo bad (very old video, but you can see buff vanish/stealth on target rogue, and how he can vanish blinds and stealth is broken):
> 
> https://www.youtube.com/watch?v=spH5R2pTBFU
> 
> 0:17 Buffs on target rogue are displayed 0:22 He cast blind AFTER enemy rogue vanishes(see buffs again) and cc is applied 0:48 He vanishes when blind is being casted (vanish breaks and he is inmunized) 1:31 Blind vanish 1:46 Vanish blind 2:07 inmune rocket gloves 2:36 Sap vanish 3:13 Vanish blind
> 
> https://www.youtube.com/watch?v=YYsi1hqGOhY
> 
> 4:15/4:51 Vanish deathcoil (vanish projectiles don't break stealth)
> 
> Akrios was a retail's great rogue, videos are totally legit

its already possible to vanish/shadowmeld spells with traveling time (death coil, ex), but spells that are instant/no travel time you need this PR https://github.com/azerothcore/azerothcore-wotlk/pull/1969

**Comment by UltraNix** [comment]
https://www.warcraftmovies.com/movieview.php?id=157699 at 2:25 u can see rogue using vanish, then the mage use Cone of Cold and the rogue is SLOW
